### PR TITLE
screencast: Send buffer transform information via PipeWire

### PR DESCRIPTION
**Pull request by columbarius**
PipeWire MR: https://gitlab.freedesktop.org/pipewire/pipewire/-/merge_requests/1423
Example client: https://github.com/obsproject/obs-studio/pull/7762

### CODE REVIEW

**Comment by columbarius** [comment]
Since this MR requires a future version of PipeWire, this needs guards to be able to build against older versions.

File: src/screencast/wlr_screencast.c [hunk 1]
```diff
@@ -263,6 +263,7 @@ static void wlr_output_handle_geometry(void *data, struct wl_output *wl_output,
 	struct xdpw_wlr_output *output = data;
 	output->make = strdup(make);
 	output->model = strdup(model);
+  output->transformation = transform;
```
Comment (emersion) [inline src/screencast/wlr_screencast.c:266]: Indentation is off here

File: src/screencast/pipewire_screencast.c [hunk 2]
```diff
@@ -419,6 +425,12 @@ void xdpw_pwr_enqueue_buffer(struct xdpw_screencast_instance *cast) {
 		logprint(TRACE, "pipewire: timestamp %"PRId64, h->pts);
 	}
 
+	struct spa_meta_videotransform *vt;
+	if ((vt = spa_buffer_find_meta_data(spa_buf, SPA_META_VideoTransform, sizeof(*vt)))) {
+    vt->transform = cast->target_output->transformation;
```
Comment (emersion) [inline src/screencast/pipewire_screencast.c:430]: Ditto

File: include/screencast_common.h [hunk 3]
```diff
@@ -179,6 +179,7 @@ struct xdpw_wlr_output {
 	int width;
 	int height;
 	float framerate;
+	int32_t transformation;
```
Comment (emersion) [inline include/screencast_common.h:182]: Using the `enum wl_output_transform` instead of `int32_t` would make it clearer what the values are.

**Comment by columbarius** [comment]
Bumped PipeWire dependency and tested with the official gstreamer plugin https://github.com/columbarius/xdg-desktop-portal-testscripts/blob/main/scripts/xdp-screencast.py.

File: src/screencast/wlr_screencast.c (hunk 1 above)
Comment (columbarius) [reply src/screencast/wlr_screencast.c]: Done

File: src/screencast/pipewire_screencast.c (hunk 2 above)
Comment (columbarius) [reply src/screencast/pipewire_screencast.c]: Done

File: include/screencast_common.h (hunk 3 above)
Comment (columbarius) [reply include/screencast_common.h]: Done

**Pull request merged**